skimage/morphology/tests/test_flood_fill.py

```diff
@@ -0,0 +1,164 @@
+
+    expected = np.array([[5, 5, 5, 5, 5, 5, 5],
+                         [5, 1, 1, 5, 2, 2, 5],
+                         [5, 1, 1, 5, 2, 2, 5],
+                         [1, 5, 5, 5, 5, 5, 3],
+                         [5, 1, 1, 1, 3, 3, 4]])
+
+    np.testing.assert_array_equal(image, expected)
+
+
+def test_inplace_float():
+    image = np.array([[0, 0, 0, 0, 0, 0, 0],
+                      [0, 1, 1, 0, 2, 2, 0],
+                      [0, 1, 1, 0, 2, 2, 0],
+                      [1, 0, 0, 0, 0, 0, 3],
+                      [0, 1, 1, 1, 3, 3, 4]], dtype=np.float32)
+
+    flood_fill(image, (0, 0), 5, inplace=True)
+
+    expected = np.array([[5., 5., 5., 5., 5., 5., 5.],
+                         [5., 1., 1., 5., 2., 2., 5.],
+                         [5., 1., 1., 5., 2., 2., 5.],
+                         [1., 5., 5., 5., 5., 5., 3.],
+                         [5., 1., 1., 1., 3., 3., 4.]], dtype=np.float32)
+
+    np.testing.assert_allclose(image, expected)
+
+
+def test_inplace_noncontiguous():
+    image = np.array([[0, 0, 0, 0, 0, 0, 0],
+                      [0, 1, 1, 0, 2, 2, 0],
+                      [0, 1, 1, 0, 2, 2, 0],
+                      [1, 0, 0, 0, 0, 0, 3],
+                      [0, 1, 1, 1, 3, 3, 4]])
+
+    # Transpose is noncontiguous
+    image2 = image[::2, ::2]
+
+    flood_fill(image2, (0, 0), 5, inplace=True)
+
+    # The inplace modified result
+    expected2 = np.array([[5, 5, 5, 5],
+                          [5, 1, 2, 5],
+                          [5, 1, 3, 4]])
+
+    np.testing.assert_allclose(image2, expected2)
+
+    # Projected back through the view, `image` also modified
+    expected = np.array([[5, 0, 5, 0, 5, 0, 5],
+                         [0, 1, 1, 0, 2, 2, 0],
+                         [5, 1, 1, 0, 2, 2, 5],
+                         [1, 0, 0, 0, 0, 0, 3],
+                         [5, 1, 1, 1, 3, 3, 4]])
+
+    np.testing.assert_allclose(image, expected)
+
+
+def test_1d():
+    image = np.arange(11)
+    expected = np.array([0, 1, -20, -20, -20, -20, -20, -20, -20, 9, 10])
+
+    output = flood_fill(image, 5, -20, tolerance=3)
+    output2 = flood_fill(image, (5,), -20, tolerance=3)
+
+    np.testing.assert_equal(output, expected)
+    np.testing.assert_equal(output, output2)
+
+
+def test_wraparound():
+    # If the borders (or neighbors) aren't correctly accounted for, this fails,
+    # because the algorithm uses an ravelled array.
+    test = np.zeros((5, 7), dtype=np.float64)
+    test[:, 3] = 100
+
+    expected = np.array([[-1., -1., -1., 100., 0., 0., 0.],
+                         [-1., -1., -1., 100., 0., 0., 0.],
+                         [-1., -1., -1., 100., 0., 0., 0.],
+                         [-1., -1., -1., 100., 0., 0., 0.],
+                         [-1., -1., -1., 100., 0., 0., 0.]])
+
+    np.testing.assert_equal(flood_fill(test, (0, 0), -1), expected)
+
+
+def test_neighbors():
+    # This test will only pass if the neighbors are exactly correct
```

Comment:
Is this statement true? My impression is that this only checks if neighbors are connected in the last dimension? 

On connectivity and the structuring element: I guess a large part of that functionality is covered by the tests for `watershed` and `local_maxima` but ideally this module would have at least some basic tests for the parameters `connectivity` and `selem`. You could just copy from the tests of `local_maxima`.

Reply:
No, it fills a column and then fills a row.  Both directions are checked.  Given the way the ravelling works, if it is going to break it will break in 2D.

Reply:
I'll admit I ascribe to a KISS style when it comes to testing in general... connectivity is checked in the docstring example and not replicated here, for example.

I'll add a basic test for `selem` but not directly testing `selem` exhaustively here, it's entirely exported to the mechanisms in `.extrema.py` which are tested separately was deliberate.

This is partially motivated by the length of our test suite, which has been an issue with CI builds in the past.

Reply:
I guess I just find that comment overly broad with potential for confusion. E.g. the test doesn't check for diagonal connections. I'd actually prefer something explicit like the examples from the docstring. Speaking of which, I testing docstrings is a nice feature to ensure that they are correct but I don't agree with the sentiment that they may replace tests. If someone changes them in the future they won't consider test coverage while doing so. I'd prefer the test coverage of a function to be independent of its docstring.

But same as above, feel free to resolve this. This is not worth blocking the PR for.